content/blog/2022-04-11-CML-runners-saving-models-1.md

```diff
@@ -0,0 +1,306 @@
+  accuracy)
+- `confusion_matrix.png`: a plot showing the classification results of our model
+- `random_forest.joblib`: the binary output of the trained model
+
+All of these files are saved to the `model` directory.
+
+```python
+from sklearn.ensemble import RandomForestClassifier
+from sklearn.metrics import plot_confusion_matrix
+import matplotlib.pyplot as plt
+import json
+import os
+import joblib
+import numpy as np
+
+# Read in data
+X_train = np.genfromtxt("data/train_features.csv")
+y_train = np.genfromtxt("data/train_labels.csv")
+X_test = np.genfromtxt("data/test_features.csv")
+y_test = np.genfromtxt("data/test_labels.csv")
+
+# Fit a model
+depth = 5
+clf = RandomForestClassifier(max_depth=depth)
+clf.fit(X_train,y_train)
+
+# Calculate accuracy
+acc = clf.score(X_test, y_test)
+print(acc)
+
+# Create model folder if it does not yet exist
+if not os.path.exists('model'):
+    os.makedirs('model')
+
+# Write metrics to file
+with open("model/metrics.txt", 'w+') as outfile:
+        outfile.write("Accuracy: " + str(acc) + "\n")
+
+# Plot confusion matrix
+disp = plot_confusion_matrix(clf, X_test, y_test, normalize='true',cmap=plt.cm.Blues)
+plt.savefig('model/confusion_matrix.png')
+
+# Save the model
+joblib.dump(clf, "model/random_forest.joblib")
+```
+
+# Train the model on a daily basis
+
+Now that we have a script to train our model and save it as a file, let’s set up
+our CI/CD to provision a runner and run the script. We define our workflow in
+`cml.yaml` and save it in the `.github/workflows` directory. This way GitHub
+will automatically go through the workflow whenever it is triggered. In this
+case the triggers are a manual run and the daily schedule.
+
+<admon type="info">
+
+The name of the workflow doesn’t matter, as long as it’s a `.yaml` and located
+in the `.github/workflows` directory. You can have multiple workflows in there
+as well. You can learn more in the
+[documentation](https://docs.github.com/en/actions/learn-github-actions/workflow-syntax-for-github-actions)
+here.
+
+</admon>
+
+```yaml
+name: CML
+on: # Here we use two triggers: manually and daily at 08:00
+  workflow_dispatch:
+  schedule:
+    - cron: '0 8 * * *'
+jobs:
+  deploy-runner:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: iterative/setup-cml@v1
+      - uses: actions/checkout@v2
+      - name: Deploy runner on EC2
+        env:
+          REPO_TOKEN: ${{ secrets.PERSONAL_ACCESS_TOKEN }}
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+        run: |
+          cml runner \
+              --cloud=aws \
+              --cloud-region=eu-west \
+              --cloud-type=t2.micro \
+              --single
+  train-model:
+    needs: deploy-runner
+    runs-on: [self-hosted]
+    timeout-minutes: 120 # 2h
+    container:
+      image: docker://iterativeai/cml:0-dvc2-base1
+    steps:
+      - uses: actions/checkout@v2
```

Comment:
```suggestion
    steps:
      - uses: actions/checkout@v2
      - uses: actions/setup-python@v2
        with:
          python-version: '3.x'
      - uses: iterative/setup-cml@v1
```

haven't tested, but should address [last comment](https://github.com/iterative/dvc.org/pull/3331/files#r846276599)